test/e2e/apimachinery/aggregator.go

```diff
@@ -337,7 +341,12 @@ func TestSampleAPIServer(f *framework.Framework, aggrclient *aggregatorclient.Cl
 			VersionPriority:      200,
 		},
 	}, metav1.CreateOptions{})
-	framework.ExpectNoError(err, "creating apiservice %s with namespace %s", "v1alpha1.wardle.example.com", namespace)
+	framework.ExpectNoError(err, "creating apiservice %s", "v1alpha1.wardle.example.com")
+	defer func() {
+		err = aggrclient.ApiregistrationV1().APIServices().Delete(context.TODO(), "v1alpha1.wardle.example.com", metav1.DeleteOptions{})
+		framework.ExpectNoError(err, "deleting apiservice %s", "v1alpha1.wardle.example.com")
+		waitAPICoherentResponse(f)
```

Comment:
this is a nit , but better move it to its own defer and add a comment on top explaining what this function does and why is needed, just for future us, this test is so big that is complex to follow ;)